app/src/test/java/com/example/create_account/CredentialsManagerTest.kt

```diff
@@ -35,4 +35,32 @@ class CredentialsManagerTest {
         val isPasswordValid = credentialsManager.isPasswordValid("!Passwd0")
         assertEquals(true, isPasswordValid)
     }
-}
\ No newline at end of file
+
+    @Test
+    fun givenUnusedEmail_whenCreateAccount_thenCreateAccount() {
+        credentialsManager.register("ANOTHer@te.st", "12345")
+        val isLoginSuccessful = credentialsManager.login("another@te.st", "12345")
+        assertTrue(isLoginSuccessful)
+    }
+
+    @Test
+    fun givenUsedEmail_whenCreateAccount_thenReturnError() {
+        val firstRegistrationResult = credentialsManager.register("another@te.st", "12345")
+        assertTrue(firstRegistrationResult)
+        val secondRegistrationResult = credentialsManager.register("another@te.st", "12345")
+        assertFalse(secondRegistrationResult)
+    }
+
+    @Test
+    fun givenWrongCredentials_whenCreateAccount_thenReturnError() {
+        val invalidEmailRegistrationResult = credentialsManager.register("abc.te.st", "12346")
+        assertFalse(invalidEmailRegistrationResult)
+    }
+
+    @Test
+    fun givenCaseDifferentEmails_whenCreateAccount_thenTreatAsSame() {
+        val result1 = credentialsManager.register("test@te.st", "password")
+        val result2 = credentialsManager.register("TEST@TE.st", "password")
+        assertFalse(result2)
```

Comment:
Use more concise function names, avoiding overly descriptive or repetitive phrases to keep them clear and meaningful